cfgov/legacy/static/hud/hud.js

```diff
@@ -0,0 +1,204 @@
+		if ( (data === null) || (data === 0) || (data === undefined) ) {
+			return false;
+		}
+		else if ( data.hasOwnProperty('error') ) {
+			return 'error';
+		}
+		else if ( !(data.hasOwnProperty('counseling_agencies')) ) {
+			return false;
+		}
+		else if ( !(data.hasOwnProperty('zip')) ) {
+			return false;
+		}
+		else {
+			return true;
+		}
+	}
+
+	return {
+		check_zip: check_zip,
+		check_hud_data: check_hud_data
+	}
+}());
+
+(function($, L) { // start jQuery capsule
+
+	var map;
+	var marker_array = [];
+	var zip_marker = null;
+
+
+	/*	get_url_zip() is a simple function to retrieve the zip variable from the URL */
+	function get_url_zip() {
+		var zip = '';
+		var keyvals = window.location.href.slice(window.location.href.indexOf('?') + 1).split('&');
+		$.each( keyvals , function(i, val) {
+			var parts = val.split('=');
+			if (parts[0] == 'zip') {
+				zip = parts[1];
+			}
+		});
+		return (cfpb_hud_hca.check_zip(zip));
+	}
+
+	/*	initialize_map() sets options and creates the map */
+	function initialize_map() {
+    L.mapbox.accessToken = mapbox_access_token,
+    map = L.mapbox.map('hud_hca_api_map_container', 'mapbox.streets')
+    .setView([40, -80], 2);
+
+    if (hud_data.counseling_agencies) {
+      update_map(hud_data);
+    }
+	}
+
+  $(document).ready(initialize_map);
+
+	/*	generate_google_map(data) takes the data and plots the markers, etc, on
+		the google map. It's called by get_counselors_by_zip(). */
+
+	function update_map(data) {
+		// reset the map
+		for (var i = 0; i < marker_array.length; i++ ) {
+			map.removeLayer(marker_array[i]);
+		}
+		marker_array = [];
+		if (zip_marker != null) {
+			map.removeLayer(zip_marker);
+		}
+		map.setZoom(2);
+		map.setView([40, -80]);
+
+		if ( cfpb_hud_hca.check_hud_data(data) === true ) {
+			var lat = data.zip.lat;
+			var lng = data.zip.lng;
+			var ziplatlng = [lat, lng];
+			var zoom = 14;
+
+			map.setZoom(zoom);
+			map.setView(ziplatlng);
+
+			var bounds = map.getBounds();
+      var xmax = -Infinity;
+      var xmin = Infinity;
+      var ymax = -Infinity;
+      var ymin = Infinity;
+
+			zip_marker = L.circle(ziplatlng, 3).addTo(map);
+
+			$.each( data.counseling_agencies, function(i, val) {
+        var lat = val.agc_ADDR_LATITUDE;
+        var lng = val.agc_ADDR_LONGITUDE
+        var position = new L.LatLng(lat, lng);
+
+        if(lat > ymax) ymax = lat;
+        if(lat < ymin) ymin = lat;
+        if(lng > xmax) xmax = lng;
+        if(lng < xmin) xmin = lng;
+
+				var number = i + 1;
+
+				if ( number < 10 ) {
+					number = '0' + number;
+				}
+
+        var icon = L.icon({
+          iconUrl: '/static/nemo/_/img/hud_gmap/agc_' + number + '.png',
+          iconAnchor: [20, 50]
+        });
+
+				var marker = new L.Marker(position, {icon: icon}).addTo(map);
+        marker_array[i] = marker;
+
+				marker.on('click', function() {
+					$(document.body).animate({'scrollTop':   $('#hud-result-' + number).offset().top }, 1000);
+				});
```

Comment:
The marker clicking functionality isn't behaving as expected. I must have broke it. I'll look into it.